internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -951,6 +968,21 @@ func createFleetServerBootstrapConfig(
 			es.TLS.CATrustedFingerprint = esCASHA256
 		}
 	}
+	if esClientCert != "" || esClientCertKey != "" {
+		if es.TLS == nil {
+			es.TLS = &tlscommon.Config{
+				Certificate: tlscommon.CertificateConfig{
+					Certificate: esClientCert,
+					Key:         esClientCertKey,
+				},
+			}
+		} else {
+			es.TLS.Certificate = tlscommon.CertificateConfig{
+				Certificate: esClientCert,
+				Key:         esClientCertKey,
+			}
+		}
+	}
```

Comment:
[NIT | Suggestion]
it can be made more concise and easier to read:

```suggestion
		if es.TLS == nil {
			es.TLS = &tlscommon.Config{}
		} 

		es.TLS.Certificate = tlscommon.CertificateConfig{
				Certificate: esClientCert,
				Key:         esClientCertKey,
		}
```